modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java

```diff
@@ -149,6 +148,9 @@ public void undeploy(String fileName) throws DeploymentException {
 
 	if (deploymentStore.containsFileName(fileName)) {
 	    File undeployingFile = new File(fileName);
+		if(fileName.contains("/tmp/carbonapps/")){
```

Comment:
I think there are some checks needs to be done before deleting the files. like file exists etc. Don't you need to check for file name *.zip. Have you tested by including some other artifacts in your car. rg: proxy api sequence and class mediator?